archunit/src/test/java/com/tngtech/archunit/core/importer/ClassFileImporterGenericClassesTest.java

```diff
@@ -0,0 +1,484 @@
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithComplexTypeParametersWithConcreteArrayBounds;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithComplexTypeParametersWithGenericArrayBounds;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithMultipleTypeParametersBoundByTypesWithDifferentBounds;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithMultipleTypeParametersWithGenericClassOrInterfaceBoundsAssignedToConcreteTypes;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithSingleTypeParameterBoundByTypeWithUnboundWildcard;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithSingleTypeParameterBoundByTypeWithWildcardWithLowerClassBound;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithSingleTypeParameterBoundByTypeWithWildcardWithLowerInterfaceBound;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithSingleTypeParameterBoundByTypeWithWildcardWithUpperClassBound;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithSingleTypeParameterBoundByTypeWithWildcardWithUpperInterfaceBound;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithSingleTypeParameterWithGenericClassBoundAssignedToConcreteClass;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithSingleTypeParameterWithMultipleSimpleClassAndInterfaceBounds;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithSingleTypeParameterWithSimpleClassBound;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithSingleTypeParameterWithSimpleInterfaceBound;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithSingleTypeParameterWithoutBound;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithThreeTypeParametersWithMultipleSimpleClassAndInterfaceBounds;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithThreeTypeParametersWithSimpleClassBounds;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithThreeTypeParametersWithoutBounds;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithTwoTypeParametersWithMultipleGenericClassAndInterfaceBoundsAssignedToConcreteTypes;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithTypeParameterWithInnerClassesWithTypeVariableBound;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithTypeParameterWithTypeVariableBound;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.ClassWithWildcardWithTypeVariableBounds;
+import com.tngtech.archunit.core.importer.testexamples.genericclasses.InterfaceParameterWithSingleTypeParameter;
```

Comment:
Yeah, I think that makes sense. I kind of wanted to keep the test class from exploding, but without the classes there is an elementary part missing for every test, you're right... I'm gonna move them into the test...

Reply:
I have inlined them all to be local classes. That way they better follow the BDD schema and it's easy to check out the given classes. In some parts I have duplicated the classes instead of making them an inner class, but I think that's fair, because in these cases it is not "semantically" the same class, i.e. no relevance, but it is a specific test fixture that should be considered independently of the other ones, even though by accident it coincides with another class at the moment. (in other words, I think there is no benefit reusing the same class for two tests, because it will likely never need to be adjusted in a way that makes sense for all test fixtures using it.)